src/search.cpp

```diff
@@ -265,10 +265,10 @@ void MainThread::search() {
       }
 
       // Vote according to score and depth
+      auto square = [](int x) { return x * x; };
```

Comment:
could this instead be std::pow(x, 2) ?

Reply:
std::pow(int, 2) is compiled into a single (floating point) multiply... 

However, reflecting about the types, it just comes to my mind that the expression might overflow when done in int. If one thread sees a draw, and the other a mate at some depth, we might pick the draw one...

Reply:
So either we test that the scores are not mates for a the voting procedure, or we can use floating point all along

Reply:
even value_known_win would overflow at depth > 21, so one needs a wider type, just special casing mate is not enough. 

Arguably, mate needs special treatment anyway (what's better 1 thread finding a mate in 5 or 10 threads a mate in 20 ? ... sometimes democracy doesn't work ;-) ).

Reply:
I now thing that the votes map should use int64_t instead of int.  VALUE_MATE is 32000 so squaring it and multiplying by depth could easily overflow int.  Do you agree?

Reply:
yes, that should work (assuming you use int64_t also as an argument to square/pow)

Reply:
Thanks!  Fixed.